library/core/src/intrinsics.rs

```diff
@@ -815,6 +815,7 @@ extern "rust-intrinsic" {
     /// This will statically either panic, or do nothing.
     ///
     /// This intrinsic does not have a stable counterpart.
+    #[rustc_const_unstable(feature = "const_maybe_assume_init", issue = "none")]
```

Comment:
For the feature gate name... we do not intend to ever stabilize these intrinsics. @oli-obk what is the plan for that case? Usually perma-unstable intrinsics just do not have a feature gate name.

Reply:
we don't stabilize the intrinsics, but we will "stabilize their constness", thus allowing them to be used in libcore/libstd from stable `const fn`

Reply:
Oh, so even if it is const-stable it cannot be used outside libstd? Curious.

Sounds like we want a feature gate collecting the three intrinsics then. Something like `const_assert_type` or so?

Reply:
Hmm, does that mean you can have a type that's `rustc_const_stable` without being `stable`? I think that will break rustdoc's rendering of `const fn`. https://github.com/rust-lang/rust/issues/79551, https://github.com/rust-lang/rust/pull/79548#discussion_r532311337

Reply:
@RalfJung 
> Sounds like we want a feature gate collecting the three intrinsics then. Something like const_assert_type or so?

I have removed the constification of the other two intrinsics (see [comments](https://github.com/rust-lang/rust/pull/79621#pullrequestreview-542661762)). Does the name `const_assert_type` still make sense or should i change it to something more specific like `const_assert_inhabited`?

Reply:
> Does the name const_assert_type still make sense or should i change it to something more specific like const_assert_inhabited?

Either is fine, I'd say.